### PR TITLE
Add random.set_random_state

### LINKED ISSUE #387: Feature request: function to set a random state
You can get the current random generator state of cupy via `cupy.random.get_random_state`, but there seems no function available to set a random state to cupy.

numpy has both `numpy.random.get_state` and `numpy.random.set_state`.

The implementation below works for now, but it depends on undocumented things and I think it's better to have an official function for it.
```
def cupy_set_random_state(random_state):
    device_id = cupy.cuda.Device().id
    cupy.random.generator._random_states[device_id] = random_state
```

### PR DESCRIPTION
**Pull request by niboshi**
Resolves #387.
Implements a feature requested.

### CODE REVIEW

File: cupy/random/generator.py [hunk 1]
```diff
@@ -439,6 +439,20 @@ def get_random_state():
...
+    """Sets the state of the random number generator for the current device.
+
+    Args:
+        state(RandomState): Random state to set for the current device.
+    """
+    if not isinstance(rs, RandomState):
+        raise TypeError(
+            'Random state must be an instance of RandomState. '
+            'Actual: {}'.format(type(rs)))
+    dev = cuda.Device()
```
Comment (okuta) [inline cupy/random/generator.py]: How about using `get_device_id()` ?

**Comment by niboshi** [comment]
Thank you reviewing. Fixed accordingly.

**Comment by okuta** [comment]
LGTM!

**Comment by kmaehashi** [comment]
Can I backport this? I want to use it in my test code. https://github.com/chainer/chainer/pull/3949/files#diff-d79652fbe7366154c45cdabbd5d1febaR14

**Pull request merged**